.github/SECURITY.md

```diff
@@ -0,0 +1,125 @@
+# Oppia Vulnerability Disclosure Process
+
+This vulnerability disclosure process describes how we accept and respond to security vulnerabilities from both Oppia developers and others. Our process follows 4 steps: Report, Investigate, Remediate, and Disclose.
+
+## Report
+
+Reporters should email [security@oppia.org](mailto:security@oppia.org) with:
+
+* A description of the problem.
+* Steps we can follow to reproduce the problem.
+* Affected versions.
+* If known, mitigations for the problem.
+
+We will respond within 3 days of the reporter’s email to acknowledge receipt of their report. Here is a template acknowledgement email:
+
+```
+Hi $REPORTER,
+
+Thank you for reporting this problem to us. We are currently investigating and will reach out sometime in the next 7 days once we have decided how to move forward or if we have any questions.
+
+Thanks,
+
+$OPPIA_SECURITY_TEAM_MEMBER
+```
+
+## Investigate
+
+Immediately upon receiving a report of a security problem, a member of Oppia’s security team (the tech leads for now) will assemble a vulnerability response team (VRT). This team should:
+
+* Include an Oppia tech lead.
+* Include developers (most likely 1-2) with expertise in the part of the app affected by the problem.
+* Include as few developers as possible to avoid premature publication of the problem.
+
+The tech lead will designate one VRT member as the VRT lead responsible for driving our response. The VRT lead will immediately (ideally within 24 hours) investigate the report and classify it as:
+
+* **Won’t fix**: The app is working as intended, the bug is not important enough for us to spend resources fixing, or the requested feature is not something we plan to implement.
+* **Bug**: The problem identified is legitimate, but it is not a security issue. It will be resolved through our normal bug-fixing process.
+* **Feature request**: The report identifies a new feature that should be implemented. It will be handled through our normal feature-development process.
+* **Low-severity vulnerability**: The report identifies a security vulnerability that does not meet the high-severity criteria. It will be resolved through our normal bug-fixing process.
+
+  For example, suppose we improperly sanitized user-provided data at the models layer such that user-provided code could be executed, but validation checks at the controller layer prevented an attacker from actually exploiting the vulnerability. This would be a security vulnerability because the models layer should not allow user-provided code to execute, but it would be low-severity because the controllers layer would prevent exploitation.
+
+* **High-severity vulnerability**: The report identifies an exploitable security vulnerability that, if exploited, could result in any of the following:
+  * (Confidentiality) Unauthorized access to personal user data.
```

Comment:
Should we remove the "user" here too?

Reply:
This language was provided by @BenHenning, so I defer to him on that question

Reply:
I think I have the same problem as before then. Are for example some fully anonymized statistical data about the usage of a topic (that are normally available just to admins) still under the category of "personal user data" or not and if not when someone gets access to these data is this a confidentiality breach?

Reply:
Maybe we should use a "could reasonably cause harm to someone or some group if exposed" standard? I think we need input from @seanlip and @BenHenning on this

Reply:
@BenHenning and I talked about this. How about:

```* (User data breach) Unauthorized access to personal user data.```

We don't necessarily think that other categories like what Vojta mention are necessarily automatically high-severity.

Reply:
@seanlip I think @vojtechjelinek's concern is over ambiguity in the phrase "personal user data"

> Are ... fully anonymized statistical data about the usage of a topic ... under the category of "personal user data" ... ?

Reply:
Here's a reference for the "standard triad" I referred to: https://github.com/ossf/secure-sw-dev-fundamentals/blob/main/secure_software_development_fundamentals.md#what-does-security-mean

Reply:
done

Reply:
@BenHenning Unauthorized disclosure of privileged information we typically call a “breach.”  What is privileged could include trade secrets, PII, or other nonpublic information.

I think breaches would normally be managed outside of a vulnerability management process, and breaches would not be classified as vulnerabilities.

If vulnerability is any state in which there is potential for loss, then the severity of vulnerability is the relative degree of the loss.  If a vulnerability could result in loss of user confidence in Oppia, or present legal risk that threatens the continuity of the organization, then it is high severity.

If there is no potential for loss or harm, then there is no vulnerability; instead you just have a bug.

I hope that is helpful.